presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergHiveMetadata.java

```diff
@@ -387,6 +423,146 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
         return TableStatisticsMaker.getTableStatistics(typeManager, constraint, handle, icebergTable);
     }
 
+    public void createView(ConnectorSession session, ConnectorTableMetadata viewMetadata, String viewData, boolean replace)
+    {
+        Map<String, String> properties = ImmutableMap.<String, String>builder()
+                .put(TABLE_COMMENT, PRESTO_VIEW_COMMENT)
+                .put(PRESTO_VIEW_FLAG, "true")
+                .put(PRESTO_VERSION_NAME, prestoVersion)
+                .put(PRESTO_QUERY_ID_NAME, session.getQueryId())
+                .put(TABLE_TYPE_PROP, ICEBERG_TABLE_TYPE_VALUE)
+                .build();
+
+        List<Column> columns = new ArrayList<>();
+        MetastoreContext metastoreContext = getMetastoreContext(session);
+        ColumnConverter columnConverter = metastoreContext.getColumnConverter();
+
+        for (ColumnMetadata column : viewMetadata.getColumns()) {
+            try {
+                HiveType hiveType = toHiveType(new HiveTypeTranslator(), column.getType());
```

Comment:
Fold line 442 and 443 into createTableObjectForViewCreation. We can pass in the ColumnConverterProvider as a parameter. Same for the HiveMetadata